geminidr/core/primitives_stack.py

```diff
@@ -170,19 +171,12 @@ def stackFrames(self, adinputs=None, **params):
         for ext in adinputs[0]:
             bytes = 0
             # Count _data twice to handle temporary arrays
-            for attr in ('_data', '_data', '_uncertainty'):
-                item = getattr(ext.nddata, attr)
-                if item is not None:
-                    # A bit of numpy weirdness in the difference between normal
-                    # python types ("float32") and numpy types ("np.uint16")
-                    try:
-                        bytes += item.dtype.itemsize
-                    except TypeError:
-                        bytes += item.dtype().itemsize
-                    except AttributeError:  # For non-lazy VAR
-                        bytes += item._array.dtype.itemsize
-            bytes += 2  #  mask always created
-            bytes_per_ext.append(bytes * np.multiply.reduce(ext.nddata.shape))
+            bytes += 2 * ext.data.dtype.itemsize
+            if ext.variance is not None:
+                bytes += ext.variance.dtype.itemsize
+
+            bytes += 2  # mask always created
+            bytes_per_ext.append(bytes * np.product(ext.shape))
```

Comment:
While looking at memory usage computation I found that 1) the final NDData object is not taken into account, so I cannot add the additional array here, and 2) it could be greatly simplified. Hope it's fine?

Reply:
With each cycle of `windowedOp`, are the output arrays stored directly in the final array or do the get created elsewhere in memory and then get copied? If it's the first, then this memory should *not* be included in the calculation. The memory limit provided is purely the space that can be used for calculations during each `windowedOp` cycle and is then freed up again at the end.

Reply:
The stacked array is stored in memory and then is set (with `set_section`) in the final output one, so at some point there is an additional array in memory.